script/draft-release/run.ts

```diff
@@ -57,30 +62,40 @@ function parseChannel(arg: string): Channel {
  * @param entries release notes for the next release
  */
 function printInstructions(nextVersion: string, entries: Array<string>) {
-  const object: any = {}
-  object[nextVersion] = entries.sort()
-
-  const steps = [
-    `Update the app/package.json 'version' to '${nextVersion}' (make sure this aligns with semver format of 'major.minor.patch')`,
-    `Concatenate this to the beginning of the 'releases' element in the changelog.json as a starting point:\n${jsonStringify(
-      object
-    )}\n`,
+  const baseSteps = [
     'Revise the release notes according to https://github.com/desktop/desktop/blob/development/docs/process/writing-release-notes.md',
-    'Commit the changes (on development or as new branch) and push them to GitHub',
+    'Lint them with: yarn draft-release:format',
+    'Commit these changes (on a "release" branch) and push them to GitHub',
     'Read this to perform the release: https://github.com/desktop/desktop/blob/development/docs/process/releasing-updates.md',
   ]
+  // if an empty list, we assume the new entries have already been
+  // written to the changelog file
+  if (entries.length === 0) {
+    printSteps(baseSteps)
+  } else {
+    const object = { [nextVersion]: entries.sort() }
+    const steps = [
+      `Concatenate this to the beginning of the 'releases' element in the changelog.json as a starting point:\n${format(
+        JSON.stringify(object),
+        {
+          parser: 'json',
+        }
+      )}\n`,
+      ...baseSteps,
+    ]
+    printSteps(steps)
+  }
+}
 
+/**
+ * adds a number to the beginning fo each line and prints them in sequence
+ */
+function printSteps(steps: ReadonlyArray<string>) {
+  console.log("Here's what you should do next:\n")
   console.log(steps.map((value, index) => `${index + 1}. ${value}`).join('\n'))
 }
 
 export async function run(args: ReadonlyArray<string>): Promise<void> {
-  try {
-    await spawn('git', ['diff-index', '--quiet', 'HEAD'])
-  } catch {
-    throw new Error(
-      `There are uncommitted changes in the working directory. Aborting...`
-    )
-  }
```

Comment:
> no longer aborts at the start if there are uncommitted changes in your local repo

@outofambit I'm curious about your thoughts on dropping this. I'm generally cool with doing so, and wondering if you have any specific scenarios in mind that prompted this. I can't quite think of any reason why there would be uncommitted changes on a release branch. 

The risk of dropping this does seem minimal -- overwriting changes to the package.json or changelog, or accidentally committing unrelated working directory changes. In both of these cases I expect issues to be caught in PR review. So I don't see a problem with this change, just want to get a sense of what your thoughts are on it.

Reply:
@kuychaco yes good question! the context here is that this check is pretty flakey and gives me (and others from what i hear) false rejections pretty often. and yep to your point it is quite uncommon to have any changes on a release branch. (now that i'm thinking about it, this check was added when we were less rigorous about using release branches, which might explain its initial utility better)